### ISSUE TITLE
[Meta] Implement MetNet-3 Model

### ISSUE THREAD

**Issue by jacobbieker** [openclimatefix/metnet#55]
We want to implement MetNet-3 (See #54). For this, we need to do:

- [ ] Add MaxViT blocks for interior of model
- [ ] Add integrating lead time into layers (i.e. FiLM, but need to make sure that the implementation we have from our MetNet-2 is correct)
- [ ] Embedding for Sparse + Dense layers
- [ ] ResNet blocks
- [ ] Train script for model that matches training setup detailed in #54 
- [ ] Topographical embedding layer
- [ ] Point-source to grid translator (from weather station to grid)

**Comment by Raahul-Singh** [comment]
@jacobbieker I'll start with the first task!

**Comment by jacobbieker** [comment]
Great! Thanks!

**Comment by Raahul-Singh** [comment]
@Raahul-Singh to check if the layer normalisation in the MaxVit and other blocks are implemented correctly.

**Comment by shermansiu** [comment]
Note: lucidrains has open-sourced the code here: https://github.com/lucidrains/metnet3-pytorch
